src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1089,6 +1061,63 @@ private TokenValidationResult ValidateToken(string token, JsonWebToken outerToke
             return tokenValidationResult;
         }
 
+        private static BaseConfiguration GetConfigurationAndRetryOnNetworkError(TokenValidationParameters validationParameters)
+        {
+            BaseConfiguration currentConfiguration;
+            try
+            {
+                currentConfiguration = validationParameters.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
+                return currentConfiguration;
+            }
+            catch (Exception ex)
+            {
+                IOException ioException;
+                if ((ioException = IsIOExceptionWithFailureInformation(ex)) != null)
+                {
+                    // Retry configuration retrieval on network failure (ServiceUnavailable or RequestTimeout).
+                    if (ioException.Data[HttpResponseConstants.StatusCode].Equals(HttpStatusCode.RequestTimeout) || ioException.Data[HttpResponseConstants.StatusCode].Equals(HttpStatusCode.ServiceUnavailable))
+                    {
+                        LogHelper.LogInformation(LogHelper.FormatInvariant(TokenLogMessages.IDX10265, ioException.Data[HttpResponseConstants.StatusCode], ioException.Data[HttpResponseConstants.ResponseContent], validationParameters.ConfigurationManager.MetadataAddress));
```

Comment:
We might want to consider logging these as warnings.

Reply:
Good point; I've added LogWarning to all cases where we were unable to obtain the Configuration. :)  Changes can be seen in this PR: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1784